src/kemal/middleware/filters.cr

```diff
@@ -74,7 +74,10 @@ module Kemal::Middleware
     property block : HTTP::Server::Context -> String
 
     def initialize(&block : HTTP::Server::Context -> _)
-      @block = ->(context : HTTP::Server::Context) { block.call(context).to_s }
+      @block = ->(context : HTTP::Server::Context) do 
+        block.call(context).to_s.to_json if context.request.headers["Content-Type"]? == "application/json"
```

Comment:
Hi Guys,
For what its worth, i agree with @asterite, there is no real need to make the `to_json` implicit, just because you set the content type doesn't mean *magic* needs to happen. Furthermore, if the content type was something other than `application/json` would you also attempt magic, i.e. `text/xml` or better still `application/x-gzip`.
My point is that this seems like the top of a slippery slope, and although doing `to_json` automatically seems reasonable on the surface, its also a tiny burden on the developer, and one they shouldn't object to IMO ;-)